packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -184,6 +184,24 @@ const TransferPanel = (): JSX.Element => {
           const { decimals } = selectedToken
           const amountRaw = utils.parseUnits(amount, decimals)
 
+          // check that a registration is not currently in progress
+          const l1RoutedAddress = (
+            await bridge.l1Bridge.l1GatewayRouter.functions.calculateL2TokenAddress(
+              selectedToken.address
+            )
+          )[0]
+          const l2RoutedAddress = (
+            await bridge.l2Bridge.l2GatewayRouter.functions.calculateL2TokenAddress(
+              selectedToken.address
+            )
+          )[0]
```

Comment:
Could we instead just query the L2 router and compare that against the token list info?
Like this we save a network query and have functionally the same validation

Reply:
If we keep both queries, we should prob dispatch the queries in parallel
```suggestion
          const l1RoutedAddress =
            bridge.l1Bridge.l1GatewayRouter.functions.calculateL2TokenAddress(
              selectedToken.address
            ).then(curr => curr[0])
          const l2RoutedAddress = 
            bridge.l2Bridge.l2GatewayRouter.functions.calculateL2TokenAddress(
              selectedToken.address
            ).then(curr => curr[0])
```

Reply:
Yep, I agree we don't need to check both. Now comparing against the l2 list address instead of looking it up